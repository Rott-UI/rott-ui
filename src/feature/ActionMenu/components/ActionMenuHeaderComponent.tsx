// React Imports
import React, {FC} from 'react'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {Item, Label} from '../..'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={COLOURS.GREY100}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
