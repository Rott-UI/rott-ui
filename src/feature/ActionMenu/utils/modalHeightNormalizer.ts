import {display} from '../../../utils'

export const modalHeightNormalizer = (
  dataLength: number,
  titleExist: boolean,
  subTitleExist: boolean,
  separatorTotalHeight: number,
  maxItem: number,
  itemHeight: number
) => {
  const showItemCount = dataLength > maxItem ? maxItem : dataLength
  const headerHeight = display.px(56)
  const listBottomPosition = display.px(93)

  return (
    ((showItemCount * itemHeight +
      (titleExist || subTitleExist ? headerHeight : 0) +
      listBottomPosition) *
      100) /
      display.percentage(100, 'height') +
    separatorTotalHeight
  )
}
