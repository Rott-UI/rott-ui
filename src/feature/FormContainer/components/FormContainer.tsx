// React Imports
import {FC, PropsWithChildren} from 'react'

// Component Imports
import {Item} from '../../Item'
import {Theme} from '../../../models'
import {COLOURS} from '../../../constants'

interface FormContainerProps extends PropsWithChildren {
  hasError?: boolean
  theme?: Theme
}

export const FormContainer: FC<FormContainerProps> = ({hasError, children, theme = 'light'}) => {
  return (
    <Item
      overflowHidden
      backgroundColor={theme === 'light' ? COLOURS.WHITE : COLOURS.GREY800}
      borderRadius={8}
      paddingTop={16}
      paddingBottom={hasError ? 0 : 8}>
      {children}
    </Item>
  )
}
