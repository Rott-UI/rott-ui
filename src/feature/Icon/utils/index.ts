// TODO: Icon size converter implementasyonu yapılması gerekmektedir.
