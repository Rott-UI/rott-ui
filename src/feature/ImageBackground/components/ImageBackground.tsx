// React Imports
import {FC} from 'react'

// React Native Imports
import {
  ImageBackground as RNImageBackground,
  ImageBackgroundProps as RNImageBackgroundProps,
  StyleSheet,
} from 'react-native'

// Component Imports
import {ImageBackgroundStyles} from '../styles'

// Feature Imports
import {getHasDynamicIslandState, getHasNotchState} from '../../../features/app/appSelector'

// Hook Imports
import {useAppSelector} from '../../../hooks'

// Package Imports
import {SafeAreaView} from 'react-native-safe-area-context'

interface ImageBackgroundProps extends RNImageBackgroundProps {
  disableSafeAreaView?: boolean
}

export const ImageBackground: FC<ImageBackgroundProps> = ({
  disableSafeAreaView,
  source,
  style,
  children,
  ...props
}) => {
  const hasDynamicIsland = useAppSelector(getHasDynamicIslandState)
  const hasNotch = useAppSelector(getHasNotchState)

  return (
    <RNImageBackground
      style={StyleSheet.flatten([ImageBackgroundStyles().defaultImageContainerStyle, style])}
      source={source}
      {...props}>
      {!disableSafeAreaView && (hasDynamicIsland || hasNotch) && <SafeAreaView edges={['top']} />}

      {children}
    </RNImageBackground>
  )
}
