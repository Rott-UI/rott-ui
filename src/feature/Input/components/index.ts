export * from './Input'
export * from './DefaultInput'
export * from './NumericInput'
export * from './PasswordInput'
export * from './IbanInput'
export * from './PhoneInput'
export * from './CVCInput'
export * from './ExpireDateInput'
export * from './CreditCardInput'
export * from './PinPasswordInput'
export * from './DateInput'
export * from './PlateNumberInput'
export * from './AmountInput'
export * from './CheckBoxInput'
export * from './SelectInput'
export * from './InputValidation'
