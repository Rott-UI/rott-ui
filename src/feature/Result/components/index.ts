export * from './Result'
export * from './ResultScreen'
