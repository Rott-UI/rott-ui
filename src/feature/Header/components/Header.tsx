// React Imports
import {FC} from 'react'

//React Native Imports
import {ViewProps} from 'react-native'

// Constant Imports
import {COLOURS} from '../../../constants'

// Component Imports
import {Content} from '../../Content'
import {Item} from '../../Item'
import {Icon, IconTypes} from '../../Icon'
import {Label} from '../../Label'
import {CommonUiProps} from '../../models'
import {Pressable} from '../../Pressable'

// Util and Lib Imports
import {goBack} from '../../../utils'
import {Image, ImageTypes} from '../../Image'
import {HeaderIconProps} from '../models'

interface HeaderProps extends ViewProps, CommonUiProps {
  back?: boolean
  title?: string
  subTitle?: string
  logo?: ImageTypes | IconTypes
  leftElement?: React.ReactElement | React.ReactNode
  leftIcon?: HeaderIconProps
  rightElement?: React.ReactElement | React.ReactNode
  rightIcon?: HeaderIconProps
  defaultBackgroundColor?: boolean
  paddingHorizontal?: number
  paddingVertical?: number
}

/**
 *
 * @param title - Eğer Icon yerine bir title gözükmesi isteniyorsa bu property kullanılabilir.
 * @param subTitle - Alt Baslik.
 * @param back - Bir önceki sayfaya dönülmek istendiğinde bu property'nin verilmesi yeterli olacaktır. Historyde kayıtlı geridönülebilir sayfa varsa otomatikman o sayfaya gidecektir.
 * @param logo - Verilen logo header'ın ortasında renderlanacaktır.
 * @param rightElement - Sağ tarafta render olacak elementi verebilirsiniz
 * @param rightIcon - Icon setlerinden herhangi birini verirseniz ekranda renderlanır
 * @param rightIconOnPress - Sağ tarafta renderlanan icon'a basıldığından gerçekleşmesi gereken bir işlem varsa bu prop kullanılabilir.
 * @param leftElement - Sol tarafta render olacak elementi verebilirsiniz
 * @param leftIcon - Icon setlerinden herhangi birini verirseniz ekranda renderlanır
 * @param leftIconOnPress - Sol tarafta renderlanan icon'a basıldığından gerçekleşmesi gereken bir işlem varsa bu prop kullanılabilir.
 * @param casal - Arka planın casal renkte olmasını istediğinizde kullanabilirsiniz.
 * @param paddingHorizontal - Yatayda kisisellestirilebilir bosluk
 * @param paddingVertical - Dikeyde kisisellestirilebilir bosluk
 *
 */
export const Header: FC<HeaderProps> = ({
  back,
  title,
  subTitle,
  logo,
  leftIcon,
  leftElement,
  rightIcon,
  rightElement,
  defaultBackgroundColor,
  backgroundColor,
  paddingHorizontal,
  paddingVertical,
}) => {
  return (
    <Content
      testID='header-test-id'
      row
      noPadding
      height={56}
      paddingHorizontal={paddingHorizontal ?? 24}
      paddingVertical={paddingVertical ?? 0}
      backgroundColor={defaultBackgroundColor ? COLOURS.GREY800 : backgroundColor}>
      <Item justifyContentCenter size='sm'>
        <Pressable
          width={40}
          height={40}
          justifyContentCenter
          alignItemsFlexStart
          onPress={(event) => {
            !!leftIcon?.onPress && leftIcon?.onPress(event)
            back && goBack()
          }}>
          {!leftElement && (leftIcon || back) && (
            <Icon
              mode={leftIcon?.mode}
              name={back ? 'CHEVRON_LEFT' : leftIcon?.name!}
              width={24}
              height={24}
              noStroke={leftIcon?.noStroke}
              strokeWidth={back ? 2 : leftIcon?.strokeWidth}
            />
          )}

          {leftElement}
        </Pressable>
      </Item>

      <Item alignItemsCenter size='md' paddingTop={16}>
        {logo && <Image testID='header-logo-test-id' width={127} height={24} name={logo} />}

        {!logo && title && (
          <Label
            testID='header-title-test-id'
            textCenter
            fontSize='xl'
            fontFamily='Markpro-Bold'
            justifyContentCenter
            variant='white'
            numberOfLines={1}
            marginTop={subTitle && -15}>
            {title}
          </Label>
        )}

        {!logo && subTitle && (
          <Label
            textCenter
            fontSize='lg'
            justifyContentCenter
            fontFamily='Markpro-Light'
            variant='white'
            numberOfLines={1}
            marginTop={10}
            testID='header-subtitle-test-id'>
            {subTitle}
          </Label>
        )}
      </Item>

      <Item alignItemsFlexEnd justifyContentCenter size='sm'>
        {!rightElement && rightIcon && (
          <Pressable
            testID='header-right-pressable-test-id'
            width={40}
            height={40}
            justifyContentCenter
            alignItemsCenter
            onPress={(event) => !!rightIcon?.onPress && rightIcon.onPress(event)}>
            <Icon
              mode={rightIcon.mode}
              testID='header-right-icon-test-id'
              name={rightIcon?.name}
              width={24}
              height={24}
              strokeWidth={1.5}
              variant='white'
            />
          </Pressable>
        )}

        {rightElement}
      </Item>
    </Content>
  )
}
