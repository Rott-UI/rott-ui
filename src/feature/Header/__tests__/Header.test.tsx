//Component Imports
import {Header} from '../components'
import {IconTypes} from '../../Icon'

//Util and Lib Imports
import {act, fireEvent, render, waitFor} from './../../../utils'

describe('Header -> Custom Component', () => {
  const TestId = {
    component: 'header-test-id',
    title: 'header-title-test-id',
    logo: 'header-logo-test-id',
    rightIcon: 'header-right-icon-test-id',
    back: 'header-back-test-id',
  }
  const IMG: {logo: IconTypes; rightIcon: IconTypes} = {
    logo: 'HGS',
    rightIcon: 'REMOVE_CIRCLE',
  }
  const headerText = 'Test Başlık'

  it('Yalnızca başlık çalışmalı ve snapshot eşleşmeli', () => {
    const rendered = render(<Header testID={TestId.component} title={headerText} />)
    expect(rendered).toMatchSnapshot()

    const {queryByTestId} = rendered

    //Title ekranda olmalı
    expect(queryByTestId(TestId.title)).toBeOnTheScreen()

    //RightIcon ekranda olmamalı
    expect(queryByTestId(TestId.rightIcon)).not.toBeOnTheScreen()

    //Logo ekranda olmamalı
    expect(queryByTestId(TestId.logo)).not.toBeOnTheScreen()
  })

  it('Yalnızca Logo çalışmalı ve snapshot eşleşmeli', () => {
    const rendered = render(<Header testID={TestId.component} logo={IMG.logo} />)

    const {queryByTestId} = rendered

    //Logo ekranda olmalı
    expect(queryByTestId(TestId.logo)).toBeOnTheScreen()

    //RightIcon ekranda olmamalı
    expect(queryByTestId(TestId.rightIcon)).not.toBeOnTheScreen()

    //Title ekranda olmamalı
    expect(queryByTestId(TestId.title)).not.toBeOnTheScreen()
  })

  it('Başlık ve Icon var ise sadece icon gorunmeli', () => {
    const {queryByTestId, getByTestId} = render(
      <Header testID={TestId.component} title={headerText} logo={IMG.logo} />
    )

    const headerElement = getByTestId(TestId.component)
    const logoElement = getByTestId(TestId.logo)
    const rightIconElement = queryByTestId(TestId.rightIcon)

    expect(rightIconElement).not.toBeOnTheScreen()
    expect(logoElement).toBeOnTheScreen()
    expect(headerElement).not.toHaveTextContent(headerText)
  })

  // TODO: bu test refactor edilecek
  // it('Proje Render Edildikten Sonra HGS Ekranı Açılmalı ve Geri Tuşuna basıldığında Tekrar HGS ekrani kapanmali', () => {
  //   const renderedApp = render(<App />)
  //   expect(renderedApp).toBeTruthy()

  //   //HGS Butonu Ekranda Olmamalı
  //   expect(renderedApp.queryByTestId('app-hgs-test-id')).not.toBeOnTheScreen()

  //   setTimeout(async () => {
  //     //HGS Butonu Ekranda Olmalı
  //     const hgs_btn = renderedApp.findByTestId('app-hgs-test-id')
  //     expect(hgs_btn).toBeOnTheScreen()

  //     //HGS Butonuna basıldığında HGS Ekranı açılmalı
  //     fireEvent(await hgs_btn, 'onPress')
  //     expect(renderedApp.findByTestId(TestId.back)).toBeOnTheScreen()

  //     //Geri butonuna basildiginda HGS ekrani kapanmali
  //     fireEvent(await renderedApp.findByTestId(TestId.back), 'onPress')
  //     expect(renderedApp.queryByTestId(TestId.back)).not.toBeOnTheScreen()
  //   }, 1500)
  // })

  //TODO: RightIcon kontrol yapılacak. isExist ve onPress
  it('RightIcon görüntülenmeli ve onPress Methodu Çalışmalı', async () => {
    const {logo, rightIcon} = IMG

    const mockOnPress = jest.fn()
    const rendered = render(
      <Header
        testID={TestId.component}
        title={headerText}
        logo={logo}
        rightIcon={{
          name: rightIcon,
          onPress: mockOnPress,
        }}
      />
    )

    const {queryByTestId, getByTestId} = rendered

    //Right Icon Ekranda Olmali
    expect(queryByTestId(TestId.rightIcon)).toBeOnTheScreen()

    //RightIcon On Press
    await act(async () => {
      await waitFor(() => {
        fireEvent.press(getByTestId(TestId.rightIcon))
      })
    })

    expect(mockOnPress).toHaveBeenCalled()
  })
})
