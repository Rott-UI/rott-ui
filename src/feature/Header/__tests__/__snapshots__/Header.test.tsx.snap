// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header -> Custom Component Yalnızca başlık çalışmalı ve snapshot eşleşmeli 1`] = `
[
  <View
    height={56}
    paddingVertical={0}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderBottomEndRadius": undefined,
        "borderBottomStartRadius": undefined,
        "borderRadius": undefined,
        "borderTopEndRadius": undefined,
        "borderTopStartRadius": undefined,
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "flexGrow": undefined,
        "gap": undefined,
        "height": 108,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 108,
        "maxWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": 46,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": 0,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
    testID="header-test-id"
  >
    <View
      justifyContentCenter={true}
      style={
        {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": "25%",
          "zIndex": undefined,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        alignItemsFlexStart={true}
        animated={false}
        collapsable={false}
        focusable={true}
        height={40}
        justifyContentCenter={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "flex-start",
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 77,
            "justifyContent": "center",
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 77,
            "maxWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 77,
            "zIndex": undefined,
          }
        }
        testID="pressable-test-id"
        width={40}
      />
    </View>
    <View
      alignItemsCenter={true}
      paddingTop={16}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 31,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": "50%",
          "zIndex": undefined,
        }
      }
    >
      <Text
        justifyContentCenter={true}
        numberOfLines={1}
        style={
          {
            "backgroundColor": undefined,
            "bottom": undefined,
            "color": "rgba(255, 255, 255, 1)",
            "flexGrow": undefined,
            "fontFamily": "Markpro-Bold",
            "fontSize": 29,
            "fontWeight": undefined,
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "letterSpacing": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "textAlign": "center",
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="header-title-test-id"
      >
        Test Başlık
      </Text>
    </View>
    <View
      alignItemsFlexEnd={true}
      justifyContentCenter={true}
      style={
        {
          "alignItems": "flex-end",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": "25%",
          "zIndex": undefined,
        }
      }
    />
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
